RoddyCore/test/de/dkfz/roddy/tools/ComplexLineTest.groovy

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2017 eilslabs.
+ *
+ * Distributed under the MIT License (license terms are at https://www.github.com/eilslabs/Roddy/LICENSE.txt).
+ */
+
+package de.dkfz.roddy.tools
+
+import groovy.transform.CompileStatic
+import org.junit.Test
+
+/**
+ * A test class for the Complex Line class
+ * Created by heinold on 16.01.17.
+ */
+@CompileStatic
+class ComplexLineTest {
+
+
+    public static final String commandLineExample = "--abc --def=\"ab,def;efk=test;anotherValue='equals'\" --jki --a=\"something=def\""
+    public static final String[] splittedCommandLineExample = [
+            "--abc",
+            "--def=\"ab,def;efk=test;anotherValue='equals'\"",
+            "--jki",
+            "--a=\"something=def\""
+    ]
+
+    public static final String codeLineExample = "def multiline='a string', b={something[in a braquet]}, method(a(b()))"
+    public static final String[] splittedCodeLineExample = [
+            "def multiline='a string'",
+            " b={something[in a braquet]}",
+            " method(a(b()))"
+    ]
+
+    public static final String faultyLineExample = "a(b(c())"
```

Comment:
I would like to see more aberrant inputs resulting in proper error messages or at least warnings (What is wrong? Which character in which string?). 

forgotten closing paren: "(blabla"
confused/non-matching paren types: "(a\\"b)\\""
forgotten opening paren: "balbla}"
non-matching paren types: "\\"blabla\\'"
singleton opening paren: "{"
singleton closing paren: ")"
etc.

Most of these inputs are wrong in 99% of the cases (even though they may not have to be fatal in the special case). But then, why should "(blabla" raise a parse error, but not "blabla)"? So if this is going to be a non-strict parser, what would be the conditions under which it should fail fatally and when to continue silently or only with a warning?